### ISSUE TITLE
report_indivudual_runs is a string not a boolean

### ISSUE THREAD

**Issue by ilent2** [EnricoMi/publish-unit-test-result-action#194]
I noticed that the action option `report_individual_runs` needs to be a string.  The example provided in README.md, shows this, but the options table does not include the quotes.

Could I suggest either changing the options table to have `"false"` and `"true"` (with quotes) or removing the quotes from the input?  When I first used this action I found this option confusing, especially since there is no warning if the user enters `true` instead of `"true"`.

**Edit:** I'm also not the most familiar with yaml, so I might have misunderstood the types.

**Comment by ilent2** [comment]
I've added two pull requests that might make it a bit clearer for first time users like me:

* The first is just an update to the documentation.  Perhaps some of the other options also need changes.
* The second adds a new helper function for parsing booleans.

Perhaps there is another option that would be better?

**Comment by EnricoMi** [comment]
Though there is typing in YAML, GitHub actions passes the values to the action through environment variables. These are strings only, so the action only retrieves strings. Here is an example

The following values in YAML are given to the action
```
Run ./
  with:
    check_name: Unit Test Results (Dockerfile)
    files: artifacts/**/*.xml
    LOG_LEVEL: DEBUG
    github_token: ***
    github_retries: 10
    ...
```
https://github.com/EnricoMi/publish-unit-test-result-action/runs/4380636910?check_suite_focus=true#step:5:217

The action runs in a docker container and retrieves the values through environment variables prefixed with `INPUT_`:
```
/usr/bin/docker run --name e28490f4807087e2365c64495c8d5587a118dd_63617f --label e28490 --workdir /github/workspace --rm -e INPUT_CHECK_NAME -e INPUT_FILES -e INPUT_LOG_LEVEL -e INPUT_GITHUB_TOKEN -e INPUT_GITHUB_RETRIES -e INPUT_COMMIT -e INPUT_COMMENT_TITLE -e INPUT_FAIL_ON -e INPUT_TIME_UNIT -e INPUT_REPORT_INDIVIDUAL_RUNS -e INPUT_DEDUPLICATE_CLASSES_BY_FILE_NAME -e INPUT_HIDE_COMMENTS -e INPUT_COMMENT_ON_PR -e INPUT_COMMENT_MODE -e INPUT_COMPARE_TO_EARLIER_COMMIT -e INPUT_PULL_REQUEST_BUILD -e INPUT_EVENT_FILE -e INPUT_EVENT_NAME -e INPUT_TEST_CHANGES_LIMIT -e INPUT_CHECK_RUN_ANNOTATIONS -e INPUT_CHECK_RUN_ANNOTATIONS_BRANCH -e INPUT_SECONDS_BETWEEN_GITHUB_READS -e INPUT_SECONDS_BETWEEN_GITHUB_WRITES -e HOME -e GITHUB_JOB ...
```
https://github.com/EnricoMi/publish-unit-test-result-action/runs/4380636910?check_suite_focus=true#step:5:217

Inside the container a Python script reads those environment variables, which are string values.

The action should work with `true` and `"true"` equally well. Can you point me to the two runs, one with `true` and one with `"true"` so I can see the difference?

**Comment by EnricoMi** [comment]
As you can see from tests, only `"true"` is allowed to set REPORT_INDIVIDUAL_RUNS to `true`:
https://github.com/EnricoMi/publish-unit-test-result-action/blob/dc4b2f22e07d0995dffe65daf902598f695e8827/python/test/test_action_script.py#L314-L322

Have you used the value `"True"` at some point?

**Comment by ilent2** [comment]
Hi Enrico, Thanks for taking a look at this.

I've taken a look back at the two runs and I think there might be something else going on that I don't understand.

The two runs are:
https://github.com/ilent2/ott/runs/4354044008?check_suite_focus=true
https://github.com/ilent2/ott/runs/4358344015?check_suite_focus=true

One uses `"true"` and the other uses `true` for the report individual runs option.  Originally I though it was just the use of `"true"`/`true` that was causing my problem, but now I've seen the problem happen regardless of if I used `"true"` or `true`.

I also think that maybe I might have made a typo in other tests, perhaps I typed `True` or `treu` instead of `true` (but I'm not sure exactly).  Perhaps it would be good to check the inputs are either `true` or `false` and throw an error or warning into the log if the user made a mistake?  Something like #196?

Best,
Isaac
